client/js/components/home/teamCreationForm.js

```diff
@@ -79,28 +81,48 @@ var TeamCreationForm = React.createClass({
         this.setState({tags: _.uniq(tags), creatingTag: false})
       },
       error: function (xhr, status, err) {
+        console.log(xhr.status + ' ' + xhr.statusText)
+        console.log(xhr.responseText)
         var message = 'Tag creation failed due to unknown reason. Try again later.'
+        if (xhr.status === HttpStatus.BAD_REQUEST) {
+          if (xhr.responseJSON && xhr.responseJSON.name) {
+            message = 'Invalid tag name: ' + xhr.responseJSON.name
+          } else {
+            message = 'Tag creation failed: ' + xhr.responseText
+          }
+        }
         this.showError(message)
       }
     })
   },
 
   // FIXME Temporary until react-select fixes allowCreate={true}
-  filterOptions: function (options, filter, currentValues) {
+  filterOptions: function (options, filterValue, currentValues) {
     // ditch existing values
     var filteredOptions = _(options)
         .difference(currentValues)
 
-    if (filter) {
+    if (filterValue) {
+      var potentialTag = filterValue.toLowerCase()
+        .replace(/[^\w-]/g, '_')
+        .replace(/__+/g, '_')
+
       // only the values matching the typed string
       filteredOptions = filteredOptions
-        .filter((o) => RegExp(filter, 'ig').test(o.label))
+        .filter((o) => RegExp(potentialTag, 'ig').test(o.label))
 
       // if the typed string doesn't exactly match an existing tag...
-      if (!filteredOptions.find((o) => o.label === filter)) {
+      if (!filteredOptions.find((o) => o.label.toLowerCase() === potentialTag)) {
         // ... add the option to create the tag
         filteredOptions = filteredOptions
-          .concat(_.some(currentValues, {label: filter}) ? [] : [{label: `Add \"${filter}\"...`, value: filter, create: true}])
+          .concat(
+              _.some(currentValues, {label: potentialTag})
+              ? []
+              : [{
+                label: `Add \"${potentialTag}\"...`,
+                value: potentialTag,
+                create: true
+              }])
       }
```

Comment:
@mvillis I'm not sure about this change... it seems to violate the principle of least surprise.

"But that's not what I typed!"